addons/mass_mailing/models/mass_mailing.py

```diff
@@ -80,102 +64,134 @@ class MassMailingContact(osv.Model):
-        }}
-
-    def create(self, cr, uid, vals, context=None):
+    name = fields.Char()
+    email = fields.Char(required=True)
+    create_date = fields.Datetime(string='Create Date')
+    list_id = fields.Many2one(
+        'mail.mass_mailing.list', string='Mailing List',
+        ondelete='cascade', required=True, default=lambda self: self.env['mail.mass_mailing.list'].search([], limit=1, order='id desc'))
+    opt_out = fields.Boolean(string='Opt Out', help='The contact has chosen not to receive mails anymore from this list')
+    unsubscription_date = fields.Datetime(string='Unsubscription Date')
+    message_bounce = fields.Integer(string='Bounce', help='Counter of the number of bounced emails for this contact.')
+
+    @api.model
+    def create(self, vals):
         if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.datetime.now() or False
-        return super(MassMailingContact, self).create(cr, uid, vals, context=context)
+            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now()
+        return super(MassMailingContact, self).create(vals)
 
-    def write(self, cr, uid, ids, vals, context=None):
+    @api.multi
+    def write(self, vals):
         if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.datetime.now() or False
-        return super(MassMailingContact, self).write(cr, uid, ids, vals, context=context)
+            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now()
+        return super(MassMailingContact, self).write(vals)
 
-    def get_name_email(self, name, context):
-        name, email = self.pool['res.partner']._parse_partner_name(name, context=context)
+    def get_name_email(self, name):
+        name, email = self.env['res.partner']._parse_partner_name(name)
         if name and not email:
             email = name
         if email and not name:
             name = email
         return name, email
 
-    def name_create(self, cr, uid, name, context=None):
-        name, email = self.get_name_email(name, context=context)
-        rec_id = self.create(cr, uid, {'name': name, 'email': email}, context=context)
-        return self.name_get(cr, uid, [rec_id], context)[0]
+    @api.model
+    def name_create(self, name):
+        name, email = self.get_name_email(name)
+        contact = self.create({'name': name, 'email': email})
+        return contact.display_name
 
-    def add_to_list(self, cr, uid, name, list_id, context=None):
-        name, email = self.get_name_email(name, context=context)
-        rec_id = self.create(cr, uid, {'name': name, 'email': email, 'list_id': list_id}, context=context)
-        return self.name_get(cr, uid, [rec_id], context)[0]
+    @api.model
+    def add_to_list(self, name, list_id):
+        name, email = self.get_name_email(name)
+        contact = self.create({'name': name, 'email': email, 'list_id': list_id})
+        return contact.display_name
```

Comment:
I agree, but you cannot change the return value type of a function.

Imagine a custom module uses this method and the return value, their 
code will be broken...

On 06/17/2016 11:14 AM, Keyur Gajjar wrote:
>
> In addons/mass_mailing/models/mass_mailing.py 
> <https://github.com/odoo/odoo/pull/9262#discussion_r67481187>:
>
> >
> > -    def add_to_list(self, cr, uid, name, list_id, context=None):
> > -        name, email = self.get_name_email(name, context=context)
> > -        rec_id = self.create(cr, uid, {'name': name, 'email': email, 'list_id': list_id}, context=context)
> > -        return self.name_get(cr, uid, [rec_id], context)[0]
> > +    @api.model
> > +    def add_to_list(self, name, list_id):
> > +        name, email = self.get_name_email(name)
> > +        contact = self.create({'name': name, 'email': email, 'list_id': list_id})
> > +        return contact.display_name
>
> return value is never used actually.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/odoo/odoo/pull/9262/files/2f17b1c454a9ff2fcd5ee6b5489a217b90ddacaf..3cdd9891cfab771475b76ae6c79c7b4a2eda0f12#r67481187>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AIC4p3sRhgSY6q2tM3RzglVZwF_OF6Mmks5qMmWHgaJpZM4GWP-T>.
>

-- 
TIVISSE Yannick
User Experience Coordinator
odoo
+32 81 81 37 00
yti@odoo.com
Chaussée de Namur, 40
1367, Grand-Rosière (BELGIUM)